### ISSUE TITLE
RangeError: Invalid string length

### ISSUE THREAD

**Issue by Uncontained0** [KatsuteDev/Background#81]
### Operating System

Windows 10

### VSCode Version

1.74.2

### Extension Version

2.3.1

### Issue

When attempting to add a large number of images as backgrounds, the extension fails on install throwing the following error in Output:

```
RangeError: Invalid string length
    at Array.join (<anonymous>)
    at getJS (c:\Users\Uncon\.vscode\extensions\katsute.code-background-2.3.1\src\extension.ts:215:44)
    at installJS (c:\Users\Uncon\.vscode\extensions\katsute.code-background-2.3.1\src\extension.ts:113:55)
    at c:\Users\Uncon\.vscode\extensions\katsute.code-background-2.3.1\src\command\install.ts:26:14
    at s.h (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:94:107622)
    at s.$executeContributedCommand (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:94:108311)
    at u.N (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:11208)
    at u.M (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10926)
    at u.H (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10019)
    at u.G (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:9000)
    at d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:7788
    at d.invoke (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
    at w.deliver (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
    at g.fire (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
    at h.fire (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
    at d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:120:15804
    at d.invoke (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
    at w.deliver (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
    at g.fire (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
    at h.fire (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
    at c.y (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:17324)
    at d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:15795
    at d.invoke (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
    at w.deliver (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
    at g.fire (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
    at g.acceptChunk (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:12045)
    at d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:11332
    at Socket.l (d:\VSCode\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:19792)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23) background.install {"value":"Katsute.code-background","c":"katsute.code-background"}
```

### How to replicate

I'm not sure the exact replication requirements, but I've narrowed this down to string length, the length of the string is exceeding the maximum string length. I'm using over 250 files each with with 6 - 12 file name lengths.

**Comment by Katsute** [comment]
Most likely caused by #69. Might be fixed by #78.

Related: #77

**Comment by Uncontained0** [comment]
The proposed fix imposes a limit of 10 for images, this isn't ideal. I'm not sure, but I think if you use a write stream or appendFile instead of writing the file as one string, you shouldn't have this issue.

**Comment by Katsute** [comment]
This particular issue was fixed in #78. The extension was writing too much information because images were being stored as a base64 string. 

The 10 cap is an additional optimization for editor images. To be clear, the extension would still store your 250 images, but the editor tabs would only store 10 at a time, since it is unlikely anyone would be using that many editor tabs at once.
